File: BlockVVV/BlockVVV/ViewController.m
//
//  ViewController.m
//  BlockVVV
//
//  Created by Songwen Ding on 6/23/16.
//  Copyright © 2016 Songwen Ding. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    
    dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
    
    for (int i = 0; i < 10; i++) {
        dispatch_async(queue, ^{
            NSLog(@"%@---%d", [NSThread currentThread], i);
        });
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: url check/url check/ViewController.swift
//
//  ViewController.swift
//  url check
//
//  Created by Songwen Ding on 2017/12/18.
//  Copyright © 2017年 Songwen Ding. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let str = "https://docs.bytedance.net/doc/oQFvgPZwtjm8vmCTrACzhb"
        print(self.validateURL(url: str))
    }

    func validateURL(url: String) -> Bool {
        let urlRegex = "(https?|ftp|file)://[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]"
        let urlPredicate = NSPredicate(format: "SELF MATCHES %@", urlRegex)
        return urlPredicate.evaluate(with: url)
    }
    func validateDocsURL(url: String) -> Bool {
        let urlRegex = "(https?)://[-A-Za-z0-9+&@#/%?=~_|!:,.;]+/(doc|sheet|folder)/+[A-Za-z0-9]{1,100}"
        let urlPredicate = NSPredicate(format: "SELF MATCHES %@", urlRegex)
        return urlPredicate.evaluate(with: url)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

